lib/openshift-rest-client.js

```diff
@@ -69,30 +70,67 @@ const spec = JSON.parse(zlib.gunzipSync(fs.readFileSync(path.join(__dirname, 'sp
+  const clientConfig = { backend: null, spec, getNames };
 
-  if (kubeconfig) {
+  if (config) {
     // A config is being passed in.  Check if it is an object
-    if (typeof kubeconfig === 'object' && kubeconfig.auth) {
+    if (typeof config === 'object' && config.auth) {
       // Check for the auth username password
-      if ('user' in kubeconfig.auth || 'username' in kubeconfig.auth) {
+      if ('user' in config.auth || 'username' in config.auth) {
         // They are trying the basic auth.
         // Get the access token using the username and password
         // Check to see if we are passing in a username/password
-        const accessToken = await getTokenFromBasicAuth({ insecureSkipTlsVerify: kubeconfig.insecureSkipTlsVerify, url: kubeconfig.url, user: kubeconfig.auth.username || kubeconfig.auth.user, password: kubeconfig.auth.password || kubeconfig.auth.pass, authUrl: kubeconfig.authUrl });
+        const { insecureSkipTlsVerify, url, authUrl } = config;
+        const user = config.auth.username || config.auth.user;
+        const password = config.auth.password || config.auth.pass;
 
-        kubeconfig = {
-          url: kubeconfig.url,
-          auth: {
-            bearer: accessToken
-          },
-          insecureSkipTlsVerify: 'insecureSkipTlsVerify' in kubeconfig ? Boolean(kubeconfig.insecureSkipTlsVerify) : false
+        const accessToken = await getTokenFromBasicAuth({ insecureSkipTlsVerify, url, user, password, authUrl });
+        const clusterUrl = authUrl || url;
+        // Create clusterName from clusterUrl by removing 'https://'
+        const clusterName = clusterUrl.replace(/(^\w+:|^)\/\//, '');
+        config = {
+          apiVersion: 'v1',
+          clusters: [
+            {
+              cluster: {
+                server: clusterUrl,
+                'insecure-skip-tls-verify': insecureSkipTlsVerify
+              },
+              name: clusterName
+            }
+          ],
+          contexts: [
+            {
+              context: {
+                cluster: clusterName,
+                user: `${user}/${clusterName}`
+              },
+              name: `default/${clusterName}/${user}`
+            }
+          ],
+          'current-context': `default/${clusterName}/${user}`,
+          kind: 'Config',
+          preferences: {},
+          users: [
+            {
+              name: `${user}/${clusterName}`,
+              user: {
+                token: accessToken
+              }
+            }
+          ]
         };
+
+        const kubeconfig = new KubeConfig();
+        kubeconfig.loadFromString(JSON.stringify(config));
+        clientConfig.backend = new Request({ kubeconfig });
       }
+    } else if (config.clusters) {
+      clientConfig.backend = new Request({ kubeconfig: config });
     }
   }
-
-  const client = new Client({ config: kubeconfig || config.fromKubeconfig(), spec, getNames: getNames });
+  const client = new Client(clientConfig);
```

Comment:
i think you still need to pass the `spec` and `getNames`  since spec is the thing that actually loads the openshift swagger api

Reply:
that's already done [here](https://github.com/nodeshift/openshift-rest-client/pull/171/files#diff-c0782fd6b5b7064b86ae2e67194217a5R74)

Reply:
ah yes.  don't mind me :)